ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSbasic.cpp

```diff
@@ -26,6 +26,22 @@ using std::string;
 
 #include "Universal_System/estring.h"
 
+namespace enigma {
+
+template <typename EffectInterface, typename EffectParams>
+void set_sound_effect_parameters(int sound, REFGUID rguidObject, DWORD dwIndex, REFGUID rguidInterface, EffectParams* effectParams) {
+  const Sound& snd = sounds.get(sound);
+
+  ComPtr<IDirectSoundBuffer8> soundBuffer8 = 0;
+  snd.soundBuffer->QueryInterface(IID_IDirectSoundBuffer8, (void**)&soundBuffer8);
+
+  ComPtr<EffectInterface> sndFX8 = 0;
+  soundBuffer8->GetObjectInPath(rguidObject, dwIndex, rguidInterface, (void**)&sndFX8);
```

Comment:
I tested this `dwIndex` and it's actually not what the DirectSound documentation makes it sound like. It's fine to always pass 0 for it in GM's case because it's actually only a search parameter that specifies the starting index.
https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ee418068(v=vs.85)

> rguidObject
Unique **_class_** identifier of the object being searched for, such as GUID_DSFX_STANDARD_ECHO. Set this parameter to GUID_All_Objects to search for objects of any class.
dwIndex
Index of the object within objects of that **_class_** in the path. See Remarks.

The DirectMusic documentation is clearer about this in that it's the index of the effect object when you have multiple effects of the same class type, which GM never does.